crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use async_channel::Receiver;
+use futures::executor::block_on;
+use mosquitto_rs::{Client, Message, QoS};
+use slight_common::BasicState;
+
+#[derive(Clone)]
+pub struct MosquittoImplementor {
+    mqtt: Arc<Mutex<Client>>,
+    host: String,
+    port: i32,
+    subscriptions: Arc<Mutex<Vec<String>>>,
+    subscriber: Arc<Mutex<Option<Receiver<Message>>>>,
+}
+
+impl std::fmt::Debug for MosquittoImplementor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "MosquittoImplementor")
+    }
+}
+
+// Pub+Sub
+impl MosquittoImplementor {
+    pub fn new(slight_state: &BasicState) -> Self {
+        let mqtt = Client::with_auto_id().unwrap();
```

Comment:
repalce `unwrap()` with a better error msg. I would actually make this function return a `anyhow::Result<Self>` and surface the error to the `pubsub::sub_open` to handle.

Reply:
I'll make these `expect`s.

Reply:
actually, I created an issue to improve the way we get configs for all crates: https://github.com/deislabs/spiderlightning/issues/191 ~ will be postponing the work 'till then, 'cause I think this same improvement could be applied in other places.